.github/workflows/test-all-packages.yml

```
@@ -515,17 +515,17 @@ jobs:
           node-version: ${{ steps.vars.outputs.node-version }}
       # END-TEST-BOILERPLATE
       - name: yarn test (SwingSet)
-        if: always()
+        if: success() || failure()
         run: cd packages/SwingSet && yarn ${{ steps.vars.outputs.test }} 'test/**/test-[E-Ie-i]*.js' | $TEST_COLLECT
       - name: yarn test (xsnap-lockdown)
-        if: always()
+        if: success() || failure()
         run: cd packages/xsnap-lockdown && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       - name: yarn test (xsnap)
-        if: always()
+        if: success() || failure()
         run: cd packages/xsnap && yarn ${{ steps.vars.outputs.test }} | $TEST_COLLECT
       # explicitly test the XS worker, for visibility
       - name: yarn test (SwingSet XS Worker)
-        if: always() && matrix.engine != 'xs'
+        if: success() || failure() && matrix.engine != 'xs'
```

What's the precedence here? Maybe the following is safer ?
```suggestion
        if: (success() || failure()) && matrix.engine != 'xs'
```

---

Agreed.